website/content/docs/connect/proxies/integrate.mdx

```diff
@@ -9,183 +9,141 @@ description: >-
+If the local Consul agent is down or unresponsive, the success rate of new connections will be compromised.
+The agent uses locally-cached data to authorize the connection and typically responds in microseconds. As a result, the TLS handshake typically spans microseconds.
 
-For a proxied service configured with a [protocol] of TCP, potentially
-long-lived TCP connections will be authorized only when they are established.
-Since many services (e.g. databases) typically use persistent connection pools,
-a change in intentions that newly denies access currently does not terminate
-existing connections in violation of the updated intention. In this case it may
-appear as if the intention is not being enforced.
+~> **Note:** This endpoint is only suitable for L4 (e.g., TCP) integration. The endpoint always treats intentions with `Permissions` defined (i.e., L7 criteria) as `deny` intentions during evaluation.
 
-Consul eventually may support a mechanism for tracking specific connections in
-the agent and then allow the agent to tell the proxy to close those connections
-when their authorization state changes, but for now that is not on the roadmap.
+The proxy can query the [intention match API](/api/connect/intentions#list-matching-intentions) endpoint, which lists intentions that match the proxy destination. The matches are represented in the native configuration of the proxy, such as RBAC for Envoy.
 
-It is recommended therefore to do one of the following:
+For performance and reliability reasons, querying the intention match API endpoint is the recommended method for implementing intention enforcement. The cached intentions should be consulted for each incoming connection (L4) or request (L7) to determine if the should be accepted or rejected.
 
-1. Have connections terminate after a configurable maximum lifetime of say
-   several hours. This balances the overhead of establishing new connections
-   while keeping an upper bound on how long after Intention changes existing
-   connections remain open.
-2. Periodically re-authorize every open connection. The AuthZ call itself is
-   not expensive and should be a local, in-memory operation so authorizing
-   thousands of open connections once every minute or so is likely to be
-   negligible overhead, but enforces a tighter upper bound on how long it takes
-   to enforce Intention changes without affecting protocol efficiency of
-   persistent connections.
+#### Persistent TCP connections and intentions
 
-#### Certificate Serial in AuthZ
+For a proxied service configured with the TCP [protocol], potentially long-lived TCP connections will only be authorized when the connections are established. But because many services, such as databases, typically use persistent connection pools, changing intentions to deny access does not terminate existing connections. This behavior violates the updated intention. In these cases, it may appear as if the intention is not being enforced.
```

Comment:
```suggestion
For a proxied service configured with the TCP [protocol], potentially long-lived TCP connections will only be authorized when the connections are initially established. But because many services, such as databases, typically use persistent connection pools, changing intentions to deny access does not terminate existing connections. This behavior violates the updated intention. In these cases, it may appear as if the intention is not being enforced.
```